Include check for extension-less input file

jadball opened the pull request:
check_input.py was unable to create a directory with the same name as the input filename if the filename did not have an extension. This is due to filesystem limitations on Linux.
This adds a check to the filename to see whether it has an extension.
If not, the suffix '_dir' is added to variable stem so that, later on, the directory is created as filename_dir

jonwright commented:
Thanks for taking a look at this. Can you give an example for which kind of filename is causing the problem? In case it is the input file, another option would be to check whether it ends with ".inp" and print an error message accordingly. In case anyone is relying on the current naming conventions I am a bit nervous about changing them...

jadball commented:
The problem occurs when your input filename doesn't have an extension.
For example:

```
$ fitallb.py -i testFABinput 
{'killfile': None, 'filename': 'testFABinput'}
testing

No absorption correction performed

No strain to stress conversion
Resume refinement
Usage: fitallb.py [options]

Options:
  -h, --help            show this help message and exit
  -i FILENAME, --input=FILENAME
                        Name of the file containing the input parameters
Traceback (most recent call last):
  File "bin/fitallb.py", line 16, in <module>
    gofitallb.run(options)
  File "lib/python2.7/site-packages/FitAllB/gofitallb.py", line 112, in run
    build_fcn.FCN(far)
  File "lib/python2.7/site-packages/FitAllB/build_fcn.py", line 599, in FCN
    f = open('./%s/fcn.py' %inp.fit['direc'],'w')
IOError: [Errno 20] Not a directory: './testFABinput/fcn.py'
```

This is likely because, in Linux, you can't make a directory with exactly the same name as a file that also exists in the same parent directory. So when check_input.py creates a directory based on the filename, it is unable to due to the current way the name of the directory is generated:

```
stem = split(self.filename,'.')[0]  
self.fit['stem'] = stem
...
if not os.path.exists(self.fit['stem']):
    os.mkdir(self.fit['stem'])
```

I have introduced a check to see whether your input file has no extension.
In that instance, the directory will be created with a '_dir' suffix to ensure that it can indeed be created on Linux.

jadball commented:
Re-reading your comment, it indeed makes more sense to do one of the following, just in case someone's workflow relies on the current output directory naming scheme:

- Only allow input filenames that end with ".inp"
- Explicity disallow input filenames that do not have an extension

jonwright commented:
I guess it is better to give an error message telling people how to fix the problem (e.g. please add a .inp extension to the filename) but leave it open for any extension in case someone comes along with an extension like .ini or .INP or .Inp etc ? Let me know if you want to merge it - rules like these can always change if needed...

jadball commented:
Sorry for the commit spam! Still learning this.

I've added an error message for an extension-less input file:
```
if not os.path.splitext(self.filename)[1] == '':
    stem = os.path.splitext(self.filename)[0]
else:
    logging.error('Input file must have an extension such as .inp')
    raise IOError, '\nInput file must have an extension such as .inp\n'
```

I've also added exception handling for directory creation as I discovered a new potential source of error.
Let's say you have two files in your working directory:

- FABinput.inp
- FABinput

If you were to run `fitallb.py -i FABinput.inp` you would pass the input file extension check, but the directory would still not be made, because another file named FABinput exists.
I've updated the directory creation code to throw an exception if there's a problem creating the directory:
```
if not os.path.isdir(self.fit['stem']):
    try:
        os.mkdir(self.fit['stem'])
    except OSError as e:
        logging.error('Error creating directory: ' + e.strerror)
        raise OSError(e)
```

jonwright commented:
Looks good to me!
Thanks
